src/clj/ataru/tarjonta_service/tarjonta_service.clj

```diff
@@ -31,14 +31,19 @@
       (assoc hakus avain haku-info)
       hakus)))
 
-(defn forms-in-use
-  [organization-service username]
-  (let [direct-organizations  (.get-direct-organizations-for-rights organization-service username [:form-edit])
-        all-organization-oids (map :oid (.get-all-organizations organization-service (:form-edit direct-organizations)))
-        in-oph-organization?  (some #{oph-organization} all-organization-oids)]
+(defn- forms-in-use
+  [cache-service organization-service username]
+  (let [direct-organizations    (.get-direct-organizations-for-rights organization-service username [:form-edit])
+        all-organization-oids   (map :oid (.get-all-organizations organization-service (:form-edit direct-organizations)))
+        in-oph-organization?    (some #{oph-organization} all-organization-oids)
+        query-organization-oids (sort (if in-oph-organization? nil all-organization-oids))
+        hakus                   (map (fn [oid] cache/cache-get-or-fetch cache-service
+                                         :forms-in-use
+                                         oid
+                                         #(client/get-forms-in-use query-organization-oids)) query-organization-oids)]
     (reduce hakus-by-form-key
             {}
-            (client/get-forms-in-use (if in-oph-organization? nil all-organization-oids)))))
```

Comment:
Tässä paluuarvo muuttuu radikaalisti jos `in-oph-organization?` on tosi. Aiemmin `client/get-forms-in-use` on tehnyt tarjontapyynnön ilman organisaatiorajauksia, mutta nyt `hakus` muuttuja on tyhjä lista.